app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageLoader.kt

```diff
@@ -62,22 +69,54 @@ class ImageLoader @Inject constructor(
         holder.itemNotUploaded()
 
         CoroutineScope(Dispatchers.Main).launch {
-            var value = false
+
+            var result : Boolean? = null
             withContext(Dispatchers.Default) {
+
                 if(mapHolderImage[holder] != image) {
                     // View holder has a new query image, terminate this query.
                     return@withContext
                 }
-                val sha1 = getSHA1(image)
+
+                val imageSHA1 = fileUtilsWrapper.getSHA1(context.contentResolver.openInputStream(image.uri))
+                var uploadedStatus = uploadedStatusDao.getUploadedFromImageSHA1(imageSHA1)
+                var sha1 = ""
+
+                if(uploadedStatus != null) {
+                    // returned from database.
+
+                    sha1 = uploadedStatus.modifiedImageSHA1
+                    if(mapHolderImage[holder] != image) {
+                        // View holder has a new query image, terminate this query.
+                        return@withContext
+                    }
+                    else if(uploadedStatus.imageResult || uploadedStatus.modifiedImageResult) {
+                        // already uploaded image.
+                        result = true
+                    }
+                    else if(uploadedStatus.lastUpdated!!.date >=
```

Comment:
This can invoke a NPE here, better to handle it gracefully

Reply:
Done.